### ISSUE TITLE
Chart: Legend is not updated with colors from palettes

### ISSUE THREAD

**Issue by tgpetrov** [telerik/nativescript-ui-feedback#819]
### Did you verify this is a real problem by searching the [NativeScript Forum](https://discourse.nativescript.org/)?
_Yes_

### Tell us about the problem
When the colors of the series are changed with new palettes. The new colors are not used in the legend.

### Which platform(s) does your issue occur on?
_iOS_

### Please provide the following version numbers that your issue occurs with:
- **NativeScript UI Chart version:** 3.8.0 
- **CLI:** 4.2.2
- **Cross-platform modules:** 4.2.0
- **Runtime(s):** 4.2.0

### Please tell us how to recreate the issue in as much detail as possible.
1. In the Legend example of the [demo-angular](https://github.com/telerik/nativescript-ui-samples-angular/blob/master/chart/app/examples/legend/chart-legend.component.html) add palettes:
```
    <Palette tkCartesianPalette seriesName="Bar">
        <PaletteEntry tkCartesianPaletteEntry fillColor="red" strokeColor="red"></PaletteEntry>
    </Palette>
    <Palette tkCartesianPalette seriesName="Line">
        <PaletteEntry tkCartesianPaletteEntry fillColor="green" strokeColor="green"></PaletteEntry>
    </Palette>
    <Palette tkCartesianPalette seriesName="Area">
        <PaletteEntry tkCartesianPaletteEntry fillColor="blue" strokeColor="blue"></PaletteEntry>
    </Palette>
</RadCartesianChart>
```
and add `seriesName` to each series to associate it with the palettes:
```
<BarSeries seriesName="Bar" ...
<LineSeries seriesName="Line" ...
<AreaSeries seriesName="Area" ...
```
2. Run the example on iOS and you will notice that the colors in the legend do not match the colors of the series

**Comment by zbranzov** [comment]
Fixed in nativescript-ui-chart@3.9.0

**Comment by MaxNaegele** [comment]
I have nativescript-ui-chart@3.10.0

And it does not update colors, I want a color for each bar!

` <Palette tkCartesianPalette>

                                <PaletteEntry tkCartesianPaletteEntry strokeColor="#2CACE1" fillColor="#2CACE1"></PaletteEntry>
                                <PaletteEntry tkCartesianPaletteEntry strokeColor="#A05BC5" fillColor="#A05BC5"></PaletteEntry>
                                <PaletteEntry tkCartesianPaletteEntry strokeColor="#8FC502" fillColor="#8FC502"></PaletteEntry>
                                <PaletteEntry tkCartesianPaletteEntry strokeColor="#FF3B3C" fillColor="#FF3B3C"></PaletteEntry>
                                <PaletteEntry tkCartesianPaletteEntry strokeColor="#476BEF" fillColor="#476BEF"></PaletteEntry>
                            </Palette>`
